### PR TITLE
Nb/feat/os encoder

### PR DESCRIPTION
**Pull request by nadavbarak**

### CODE REVIEW

**Comment by shlomieYaniv** [comment]
Please make `mvn spotless:apply` and commit again :)

File: src/main/java/com/whitesource/remediation/encoder/Encode.java [hunk 1]
```diff
@@ -14,20 +11,30 @@
...
+
+  /**
+   * Encodes any non alphaNumeric character that is not part of charsToIgnore.
+   *
+   * @param param An argument or part of an argument for the operating systems command.
+   * @param charsToIgnore Array of characters to not encode.
+   * @return Encoded parameter.
+   */
+  public static String forOsCommand(@NonNull final String param,  char[] charsToIgnore) {
+    StringBuilder sb = new StringBuilder();
```
Comment (shlomieYaniv) [inline src/main/java/com/whitesource/remediation/encoder/Encode.java:32]: Cool idea to use `StringBuilder`!

**Review by shlomieYaniv — CHANGES_REQUESTED** [review]
Loved the overloading 👍

File: src/main/java/com/whitesource/remediation/encoder/Encode.java [hunk 2]
```diff
@@ -43,7 +50,7 @@ public static String OSParameterEncoder(@NonNull final String params)
     for (Object content : contents) {
       results.add(logContentEncoder(content));
     }
-    return results.toArray(String[]::new);
+    return (String[]) results.toArray();
```
[the remark this answers is not in the document]
Comment (nadavbarak) [reply src/main/java/com/whitesource/remediation/encoder/Encode.java:53]: Marked as error on IDE

File: src/main/java/com/whitesource/remediation/encoder/Encode.java [hunk 3]
```diff
@@ -182,4 +189,24 @@ public static String forJavaScriptAttribute(@NonNull final String content) {
...
     return org.owasp.encoder.Encode.forJavaScriptAttribute(content);
   }
+
+  private static String encodeCharacterForOsCommand(char c, char[] charsToIgnore) {
+    boolean isAlphaNumeric = !((c < '0' || c > '9') && (c < 'A' || c > 'Z') && (c < 'a' || c > 'z'));
+    if (containsCharacter(charsToIgnore, c) || isAlphaNumeric) {
+      return ""+ c;
+    }
+
+    if (System.getProperty("os.name").toLowerCase().contains("win")) {
```
[the remark this answers is not in the document]
Comment (nadavbarak) [reply src/main/java/com/whitesource/remediation/encoder/Encode.java:199]: Requires an additional dependency of apache

**Pull request merged**